rmw_opensplice_cpp/src/qos.cpp

```diff
@@ -189,6 +201,35 @@ bool get_datawriter_qos(
   return set_entity_qos_from_profile(qos_profile, datawriter_qos);
 }
 
+bool get_topic_qos(
+  DDS::DomainParticipant & participant,
+  const rmw_qos_profile_t & qos_profile,
+  DDS::TopicQos & topic_qos)
+{
+  bool ret = true;
+
+  DDS::ReturnCode_t status;
+  status = participant.get_default_topic_qos(topic_qos);
+  if (nullptr != check_get_default_topic_qos(status)) {
+    RMW_SET_ERROR_MSG(check_get_default_topic_qos(status));
+    return false;
+  }
+
+  if (qos_profile.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL) {
+    topic_qos.durability.kind = DDS::TRANSIENT_LOCAL_DURABILITY_QOS;
+    topic_qos.durability_service.service_cleanup_delay = {0, 0};
+    ret = rmw_qos_to_history(
+                  qos_profile,
+                  topic_qos.durability_service.history_kind,
+                  topic_qos.durability_service.history_depth);
```

Comment:
nitpick: same as above, according to our style guide the indention should only be single steps of (in this case) two spaces